ui/src/app/streams/stream-deploy/stream-deploy.component.ts

```diff
@@ -93,64 +385,76 @@ export class StreamDeployComponent implements OnInit, OnDestroy {
-            this.propertiesAsMap[prop.substr(0, index)] = prop.substr(index + 1);
-          }
-        }
-      }
-    }
+  setAppProperties(config: any, app: any) {
+    const builderAppsProperties = config.builderAppsProperties;
+    const version = config.builder.get('appsVersion').get(app.name).value || app.version;
+    const options = builderAppsProperties[app.name] ? builderAppsProperties[app.name] : app.options;
+    const modal = this.bsModalService.show(StreamDeployAppPropertiesComponent);
 
-    if (this.skipperEnabled) {
-      this.propertiesAsMap['spring.cloud.dataflow.skipper.platformName'] = this.deploymentPlatform.value;
-    }
+    modal.content.title = `Properties for ${app.name} (${version})`;
+    const appPropertiesSource = new AppPropertiesSource(Object.assign([], options
+      .map((property) => Object.assign({}, property))));
 
-    const busy = this.streamsService.deployDefinition(this.id, this.propertiesAsMap)
-    .pipe(takeUntil(this.ngUnsubscribe$))
-    .subscribe(
-      data => {
-        this.toastyService.success('Successfully deployed stream definition "'
-          + this.id + '"');
-        this.router.navigate(['streams/definitions']);
-      },
-      error => {
-        if (error.message != null) {
-          this.toastyService.error(error.message);
-        } else {
-          this.toastyService.error(error);
-        }
-      }
-    );
-    this.busyService.addSubscription(busy);
+    appPropertiesSource.confirm.subscribe((properties: Array<any>) => {
+      builderAppsProperties[app.name] = properties;
+    });
+    modal.content.setData(appPropertiesSource);
+  }
+
+  getAppProperties(builderAppsProperties: {}, appId: string): Array<{ key: string, value: string }> {
+    const appProperties = builderAppsProperties[appId];
+    if (!appProperties) {
+      return [];
+    }
+    return appProperties.map((property: Properties.Property) => {
+      return (property.value !== undefined && property.value.toString() !== '' && property.value !== property.defaultValue) ? ({
+        key: `${property.name}`,
+        value: property.value
+      }) : null;
+    }).filter((app) => app !== null);
   }
 
   /**
-   * Used to read the deployment properties of a stream from a flat file
-   * @param event The event from the file selection dialog.
+   * Export the builder to a TXT file
+   *
+   * @param value
    */
-  displayFileContents(event: any) {
-    const file: File = event.target.files[0];
-    const reader: FileReader = new FileReader();
-    const _form = this.form;
-    reader.onloadend = () => {
-      _form.patchValue({deploymentProperties: reader.result});
-    };
-    reader.readAsText(file);
+  exportBuilderToFile(value: { builder: FormGroup, builderAppsProperties: any, file: FormGroup, config: StreamDeployConfig }) {
+    const propertiesString = this.toRaw(value).map((a) => `${a.key}=${a.value}`).join('\n');
+    const filename = `${value.config.id}.${new Date().getTime()}.txt`;
```

Comment:
Could we use `momentjs` to generate a more user-friendly time (ISO 8601 format)? A unix timestamp is a bit hard on the eye ;-)

E.g.:   `test_2018-03-14T162355.txt`